proofs/tree.go

```diff
@@ -287,26 +288,95 @@ func NewSalt() (salt []byte) {
 // random values.
 //
 // This method will fail if there are any fields of type other than []byte (bytes in protobuf) in the
-// message.
-func FillSalts(message proto.Message) (err error) {
-	v := reflect.ValueOf(message).Elem()
+// message leaf.
+func FillSalts(dataMessage, saltsMessage proto.Message) (err error) {
```

Comment:
I'd extend the above comments with the particulars about the Salts. Doesn't this also need to have the salts field name option or would it get filled anyway?

Reply:
ok, I will add that. Regarding the field name, it will fill every field so there is no need of explicitly providing the field name.